x/claims/keeper/claim.go

```diff
@@ -158,26 +159,25 @@ func (k Keeper) MergeClaimsRecords(
 
 	// iterate over all the available actions and claim the amount if
 	// the recipient or sender has completed an action but the other hasn't
-	for i := int32(1); i < int32(len(types.Action_value)); i++ {
-		senderCompleted := senderClaimsRecord.ActionsCompleted[i-1]
-		recipientCompleted := recipientClaimsRecord.ActionsCompleted[i-1]
 
-		action := types.Action(i)
+	for _, action := range actions {
+		senderCompleted := senderClaimsRecord.HasClaimedAction(action)
+		recipientCompleted := recipientClaimsRecord.HasClaimedAction(action)
 
 		switch {
 		case senderCompleted && recipientCompleted:
 			// both sender and recipient completed the action. No-op
-			mergedRecord.ActionsCompleted[i-1] = true
+			mergedRecord.ClaimAction(action)
```

Comment:
The name `ClaimAction(action)` sounds like the amount might also be transferred with that function. What about renaming to `MarkClaimed(action)`

Reply:
or `MarkCompleted`